### PR TITLE
Flight Task Avoidance Interface

### PR DESCRIPTION
**Pull request by mrivi**
Same as #9937, on top of current master.

Could we get this in?

### CODE REVIEW

**Comment by Stifael** [comment]
flash issue

**Comment by mrivi** [comment]
Since #10047 got merged, I have removed the FlightTask Orbit from the fmu-v2 target.
Missing flash reduced from 840 bytes to 48 bytes

**Comment by Stifael** [comment]
It probably would make sense to remove FlightTaskSport from the core tasks and adjust it to the custom Flighttask interface.

**Comment by mrivi** [comment]
We got enough flash (CI is passing). I tested it locally and not having ninja installed makes a difference.

**Comment by dagar** [comment]
This looks fine to me. Testing?

File: src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp [hunk 1]
```diff
@@ -5,7 +5,17 @@
...
 // First index of empty_setpoint corresponds to time-stamp and requires a finite number.
 const vehicle_local_position_setpoint_s FlightTask::empty_setpoint = {0, NAN, NAN, NAN, NAN, NAN, NAN, NAN, NAN, NAN, NAN, NAN, {NAN, NAN, NAN}};
+
 const vehicle_constraints_s FlightTask::empty_constraints = {0, NAN, NAN, NAN, NAN, NAN, NAN, NAN, vehicle_constraints_s::GEAR_KEEP, {}};
+const vehicle_trajectory_waypoint_s FlightTask::empty_trajectory_waypoint = {0, 0, {0, 0, 0, 0, 0, 0, 0},
+	{	{0, {NAN, NAN, NAN}, {NAN, NAN, NAN}, {NAN, NAN, NAN}, NAN, NAN, false, {0, 0, 0}},
+		{0, {NAN, NAN, NAN}, {NAN, NAN, NAN}, {NAN, NAN, NAN}, NAN, NAN, false, {0, 0, 0}},
+		{0, {NAN, NAN, NAN}, {NAN, NAN, NAN}, {NAN, NAN, NAN}, NAN, NAN, false, {0, 0, 0}},
+		{0, {NAN, NAN, NAN}, {NAN, NAN, NAN}, {NAN, NAN, NAN}, NAN, NAN, false, {0, 0, 0}},
+		{0, {NAN, NAN, NAN}, {NAN, NAN, NAN}, {NAN, NAN, NAN}, NAN, NAN, false, {0, 0, 0}}
```
Comment (dagar) [inline src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp]: You could create a constexpr empty `trajectory_waypoint` to use here.

**Comment by mrivi** [comment]
This is the branch used to test the avoidance at Auterion. Probably @baumanta can provide logs if necessary.

File: src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp (hunk 1 above)
Comment (mrivi) [reply src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp]: `empty_trajectory_waypoint` is defined as `static const vehicle_trajectory_waypoints_s` here https://github.com/PX4/Firmware/pull/10013/files/3776000147e3cdebf303ac217133649c3ab3add4#diff-e09a66ab293d2aa0badec8b7f358bf6dR133
does it make any difference declaring it as `static const` or `constexpr`?  I have read there is no difference, at least for integers.

**Comment by mrivi** [comment]
@dagar can we get this in?

File: src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp (hunk 1 above)
Comment (dagar) [reply src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp]: The point was to avoid the repetition.

**Pull request merged**